### DISCUSSION TITLE
Should v5 Signer API use micro-starknet's Signature type directly?

### DISCUSSION THREAD

**Discussion by yoga-braavos** [seanjameshan/starknet.js#565] [General]
Currently the return value from several Signer interface APIs are defined as `Signature`, which is:
`declare type Signature = _noble_curves_abstract_weierstrass.SignatureType;`

I would suggest that the Signer interface will be generic and not coupled to a specific library's types (i.e. `micro-starknet`) - As an interface it should support other use-cases that might not use or even be incompatible with micro-starknet.

I encountered this issue when migrating Braavos's code-base to starknet.js v5 and indeed our use case is incompatible with the current `Signature` type.

**Comment by dhruvkelawala** [comment]
Yes, it was bad on my side to use the Signature Class from micro-starknet. I had a discussion with @janek26 few months ago, where we faced the same issue as you. We should keep the type of Signature similar to starknet v4. i.e

`type Signature = string[]`

**Reply by tabaktoni** [reply]
If you do your own implementation and use only the interface this should be fine, but if you use the default implementation then we need a way to skip the usage of signatureToHexArray & signatureToDecimalArray?

**Comment by tabaktoni** [comment]
@yoga-braavos  @dhruvkelawala @janek26 Please take a look at this resolution. After playing around I manage to make this solution based on default implementation and to allow custom implementation without changing the interface.
https://github.com/0xs34n/starknet.js/pull/570

**Comment by ivpavici** [comment]
#570 merged, so I'll close this discussion! thanks to all!
